### PR TITLE
feat: implement config option interval

### PR DESCRIPTION
**Pull request by GregoryAlbouy**
## Changes

- Removed function `config.New` in favor to method `config.Config.WithURL`

### CODE REVIEW

**Review by moreirathomas — COMMENTED** [review]
I just wrote some of my thoughts. Otherwise it seems to do the trick. Maybe we'd like to do some unit test on the sleep behavior ?

File: cmd/runner/main.go [hunk 1]
```diff
@@ -35,9 +36,10 @@ var defaultConfigFiles = []string{
 func parseArgs() {
 	flag.StringVar(&configFile, "configFile", configfile.Find(defaultConfigFiles), "Config file path")
 	flag.StringVar(&uri, "url", "", "Target URL to request")
-	flag.IntVar(&concurrency, "concurrency", 0, "Number of connections to run concurrently")
-	flag.IntVar(&requests, "requests", 0, "Number of requests to run, use duration as exit condition if omitted")
 	flag.DurationVar(&timeout, "timeout", 0, "Timeout for each HTTP request")
+	flag.IntVar(&requests, "requests", 0, "Number of requests to run, use duration as exit condition if omitted")
+	flag.IntVar(&concurrency, "concurrency", 0, "Number of connections to run concurrently")
+	flag.DurationVar(&interval, "interval", 0, "Minimum duration between two groups of requests")
```
[the remark this answers is not in the document]
Comment (GregoryAlbouy) [reply cmd/runner/main.go:42]: Indeed "groups" is wrong and misleading about the actual implementation. "Minimum duration between two non-concurrent requests" would be more accurate.

**Comment by GregoryAlbouy** [comment]
> Maybe we'd like to do some unit test on the sleep behavior ?

Unit tests based on http requests and durations, sounds like a lot of fun 😂  Will do my best!

File: requester/requester.go [hunk 2]
```diff
@@ -71,12 +71,13 @@ func (r *Requester) Run() (Report, error) {
 		numWorker   = r.config.RunnerOptions.Concurrency
 		maxIter     = r.config.RunnerOptions.Requests
 		timeout     = r.config.RunnerOptions.GlobalTimeout
+		interval    = time.Second
```
Comment (GregoryAlbouy) [inline requester/requester.go:74]: 🤦‍♂️

**Review by moreirathomas — APPROVED** [review]

**Review by c1-ra — APPROVED** [review]

**Pull request merged**